File: service/src/main/java/com/bnorm/barkeep/service/IngredientService.java
// Copyright 2016 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.service;

import java.util.Collection;

import com.bnorm.barkeep.model.Ingredient;

public interface IngredientService {

  Collection<Ingredient> getIngredients();

  Ingredient getIngredient(long ingredientId);

  Ingredient createIngredient(Ingredient ingredient);

  Ingredient setIngredient(long ingredientId, Ingredient ingredient);

  void deleteIngredient(long ingredientId);
}

File: service/src/main/kotlin/com/bnorm/barkeep/service/IngredientService.kt
// Copyright 2017 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.service

import com.bnorm.barkeep.model.Ingredient
import com.bnorm.barkeep.model.IngredientSpec

interface IngredientService {

  fun getIngredients(): Collection<Ingredient>

  fun getIngredient(id: Long): Ingredient

  fun createIngredient(ingredient: IngredientSpec): Ingredient

  fun setIngredient(id: Long, ingredient: IngredientSpec): Ingredient

  fun deleteIngredient(id: Long)
}
